<?xml version="1.0"?>
<launch>
<arg name="world_frame"       default="map"/>
<arg name="robot_name"       default="limo"/>
<arg name="lang"              default="eng"/>
<arg name="waypoints_file" default="$(find orne_waypoints_editor)/waypoints_cfg/waypoints.yaml"/>
<arg name="amcl_filename" default="$(find fulanghua_waypoints_nav)/param/initial_pose.yaml"/>
<arg name="voice_path" default="$(find fulanghua_waypoints_nav)/voice/$(arg lang)/"/>
<arg name="dist_err"       default="0.8"/>
<arg name="cmd_vel"         default="cmd_vel"/>
<arg name="max_vel"         default="0.4"/>
<arg name="min_vel"         default="0.1"/>
<arg name="volume"         default="2.0"/>
<arg name="Kp"       default="0.4"/> 
<arg name="offset_fixed_x" default="0.009"/>
<arg name="offset_fixed_y" default="0.02"/>
<arg name="offset_fixed_z" default="0.255"/>
<arg name="image_topic" default="/usb_cam2/image_raw"/>
<arg name="calibration_path" default="$(find camera_pkg)/param"/> 

<node name="waypoints_nav" pkg="fulanghua_waypoints_nav" type="waypoints_nav" output="screen">
  <param name="world_frame" value="$(arg world_frame)"/>
  <param name="filename" value="$(arg waypoints_file)"/>
  <param name="amcl_filename" value="$(arg amcl_filename)"/>
  <param name="cmd_vel"  value="$(arg cmd_vel)" />
  <param name="robot_name"  value="$(arg robot_name)" />

</node>
  
<node name="action_server" pkg="fulanghua_action" type="adv_action_server" output="screen">
    <param name="dist_err" value="$(arg dist_err)"/> 
    <param name="cmd_vel"  value="$(arg cmd_vel)" />
    <param name="max_vel"  value="$(arg max_vel)" />
    <param name="min_vel"  value="$(arg min_vel)" />
    <param name="voice_path"  value="$(arg voice_path)" />
    <param name="volume"  value="$(arg volume)" />
</node>
<node name="rotate_server" pkg="fulanghua_action" type="rotate_action_server" output="screen">
    <param name="dist_err" value="$(arg dist_err)"/> 
    <param name="cmd_vel"  value="$(arg cmd_vel)" />
    <param name="max_vel"  value="$(arg max_vel)" />
    <param name="min_vel"  value="$(arg min_vel)" />
    <param name="voice_path"  value="$(arg voice_path)" />
    <param name="volume"  value="$(arg volume)" />
    <param name="Kp"  value="$(arg Kp)" />
</node>
  
<node name="ar_detect_server" pkg="fulanghua_action" type="aruco_detect_server" output="screen">
    <param name="cmd_vel"  value="$(arg cmd_vel)" />
    <param name="image_topic" value="$(arg image_topic)"/>
    <param name="Kp"  value="$(arg Kp)" />
    <param name="offset_fixed_x" value="$(arg offset_fixed_x)"/>
    <param name="offset_fixed_y" value="$(arg offset_fixed_y)"/>
    <param name="offset_fixed_z" value="$(arg offset_fixed_z)"/>
    <param name="calibration_path" value="$(arg calibration_path)/camera_ext.yaml" />
</node>

<include file="$(find sound_play)/soundplay_node.launch"></include>

</launch>
